### PR TITLE
IDX:2698 check group permissions

### PR DESCRIPTION
**Pull request by cgundy**
This PR adds another compliance step to check that the team listed as the repository owners in CODEOWNERS also has correct permissions set for the repository (minimum write).

We will be able to test this more comprehensively with end-to-end tests, but for now I just added a line to `open-repositories.txt` and tested the workflow manually:

Screenshot 2023-03-21 at 16 20 08

### CODE REVIEW

File: custom_python_actions/check_compliance.py [hunk 1]
```diff
@@ -38,28 +46,81 @@ def check_readme(repo: github3.github.repo) -> bool:
         return False
 
 
+def get_team_name(code_owners: str, org_name: str) -> str:
+    code_owner_file = code_owners.strip("\n")
+    regex_output = re.findall(
+        # If anyone is a regex wizard, please help me out here!
+        r"(?=\*. *@).*(@\S*).*(@\S*)|(?=\*. *@).*(@\S*)",
```
Comment (levshaATdfinity) [inline custom_python_actions/check_compliance.py:53]: It looks too complicated to me and I'm afraid will also wrongly catch cases like this:
```
# * @group/user # uncomment when ready
```

File: tests/test_check_compliance.py [hunk 2]
```diff
@@ -100,6 +120,55 @@ def test_check_readme_other_error():
         readme = check_readme(repo)
 
 
+code_owners_test_file_1 = "* @dfinity/idx\n"
```
Comment (levshaATdfinity) [inline tests/test_check_compliance.py:123]: Maybe just have them all as separate files in test_data/ for uniformity

File: custom_python_actions/check_compliance.py [hunk 3]
```diff
@@ -38,28 +46,81 @@ def check_readme(repo: github3.github.repo) -> bool:
...
+    print(f"Codeowners found {regex_output}")
+    if len(regex_output) == 0:
+        print(
+            "No repo-level team owner found. Double check the format of your CODEOWNERS file."
+        )
+        sys.exit(1)
+    team_handle = [a for a in regex_output[0] if a != ""]
+    if len(team_handle) > 1:
+        print("Only one team can be listed for repo-level codeowners.")
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (cgundy) [reply custom_python_actions/check_compliance.py:65]: If this errors, then the `check_compliance` step will fail. I think this makes it a little easier for users to debug and understand what the issue is, vs if this just succeeded and the `permissions_check` failed without explaining why.

File: custom_python_actions/check_compliance.py (hunk 1 above)
Comment (cgundy) [reply custom_python_actions/check_compliance.py]: For reference, this is where I was trying things out https://regex101.com/r/q03Bd4/1 and this was my test file:

```
* @dfinity/idx
* @dfinity/idx    # some comment
*       @global-owner1 @global-owner2
*.js    @js-owner #This is an inline comment.
*/all-files @sdk-team
* @another-org/another-team # some comment
* @dfinity/idx  @dfinity/another-team
```

File: custom_python_actions/check_compliance.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (levshaATdfinity) [reply custom_python_actions/check_compliance.py:66]: Let's go ahead with this one and unify the overall logaic as a follow up change.

**Review by levshaATdfinity — APPROVED** [review]

**Pull request merged**